server/src/main/java/org/elasticsearch/cluster/routing/ShardRoutingState.java

```diff
@@ -10,25 +10,36 @@
 package org.elasticsearch.cluster.routing;
 
 /**
- * Represents the current state of a {@link ShardRouting} as defined by the
- * cluster.
+ * Tracks the current assignment status of a particular shard copy ({@link ShardRouting}).
  */
 public enum ShardRoutingState {
     /**
-     * The shard is not assigned to any node.
+     * The shard is not assigned to any node; any data which it contains is unavailable to the cluster.
+     *
+     * A shard transitions from {@link #UNASSIGNED} to {@link #INITIALIZING} when the master wants an assigned data node to create or start
+     * recovering this shard copy. A shard may also transition back to {@link #UNASSIGNED} in case of an assignment failure.
```

Comment:
Really any failure puts it back to `UNASSIGNED` - the phrase "assignment failure" indicates to me a problem in allocation (the process which assigns shards to nodes) but in fact most unassignments will be due to other failures.

```suggestion
     * recovering this shard copy. A shard may also transition back to {@link #UNASSIGNED} in case of a failure.
```

Reply:
I fiddled with this a little more. If it still suggests allocation failures to you, I'll change it back to simply "a failure"